app/src/main/res/layout/fragment_favorite.xml

```diff
@@ -106,15 +108,29 @@
 				android:layout_width="wrap_content"
 				android:layout_height="wrap_content"
 				android:layout_margin="8dp"
+				android:layout_marginEnd="7dp"
 				android:fontFamily="serif"
 				android:padding="2dp"
 				android:textColor="@color/textColor"
 				android:textSize="16sp"
 				android:textStyle="bold"
 				app:layout_constraintBottom_toBottomOf="@id/favChipGroup"
-				app:layout_constraintEnd_toEndOf="parent"
+				app:layout_constraintEnd_toStartOf="@+id/clearAllButton"
 				app:layout_constraintTop_toTopOf="@id/favChipGroup" />
 
+			<ImageButton
+				android:id="@+id/clearAllButton"
+				android:layout_width="40dp"
+				android:layout_height="40dp"
```

Comment:
using wrap_content would do, since we aren't using scaletype=fitxy, the extra space is filled in as padding/margin; which is unnecessary in our case here.